packages/wallets/src/components/Base/WalletPasswordField/WalletPasswordField.tsx

```diff
@@ -1,15 +1,31 @@
 import React, { useCallback, useState } from 'react';
-import { Score, validatePassword, validPassword } from '../../../utils/password';
+import { Score, calculateScore, validPassword, isPasswordValid, passwordKeys } from '../../../utils/password';
 import { WalletTextField } from '../WalletTextField';
-import { WalletTextFieldProps } from '../WalletTextField/WalletTextField';
 import PasswordMeter from './PasswordMeter';
 import PasswordViewerIcon from './PasswordViewerIcon';
 import './WalletPasswordField.scss';
+import { passwordErrorMessage, passwordRegex, warningMessages } from '../../../constants/password';
+import { zxcvbn, zxcvbnOptions } from '@zxcvbn-ts/core';
+import { dictionary } from '@zxcvbn-ts/language-common';
+import { WalletPasswordFieldProps } from '../WalletPasswordFieldLazy/WalletPasswordFieldLazy';
 
-interface WalletPasswordFieldProps extends WalletTextFieldProps {
-    password: string;
-    shouldDisablePasswordMeter?: boolean;
-}
+export const validatePassword = (password: string) => {
+    const score = calculateScore(password);
+    let errorMessage = '';
+
+    const options = { dictionary: { ...dictionary } };
+    zxcvbnOptions.setOptions(options);
+
+    const { feedback } = zxcvbn(password);
+    if (!passwordRegex.isLengthValid.test(password)) {
+        errorMessage = passwordErrorMessage.invalidLength;
+    } else if (!isPasswordValid(password)) {
+        errorMessage = passwordErrorMessage.missingCharacter;
+    } else {
+        errorMessage = warningMessages[feedback.warning as passwordKeys] || '';
```

Comment:
A nitpick

```suggestion
        errorMessage = warningMessages[feedback.warning as passwordKeys] ?? '';
```

Reply:
updated (though this code was just moved from different place :) )